source/PlayerInfo.cpp

```diff
@@ -1042,11 +1052,36 @@ void PlayerInfo::BuyShip(const Ship *model, const string &name, bool isGift)
 			for(const auto &it : model->Outfits())
 				stock[it.first] -= it.second;
 		}
+
+		DiscoverOutfits(model->Outfits());
 	}
 }
 
 
 
+void PlayerInfo::DiscoverOutfits(const std::map<const Outfit *, int> &outfits)
+{
+	for(const auto &it : outfits)
+		if(!OutfitIsKnown(*it.first))
```

Comment:
Since this is set the elements are unique already, you don't need to check + insert, you can simply insert.

Reply:
will take a look